### PR TITLE
More idiomatic Rust

### PR DESCRIPTION
**Pull request by nzig**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -7,12 +7,10 @@ fn next_workspace_number(conn: &mut swayipc::Connection) -> Result<i32, swayipc:
...
     ids.sort_unstable();
-    for i in 1..ids.len() + 1 {
-        if ids[i - 1] != i as i32 {
-            return Ok(i as i32);
-        }
-    }
-    Ok((ids.len() + 1) as i32)
+    Ok(ids
+        .windows(2)
+        .find(|w| w[0] + 1 != w[1])
```
Comment (pblkt) [inline src/main.rs:12]:
```suggestion
        .find(|(cur, next)| cur + 1 != next)
```

File: src/main.rs [hunk 2]
```diff
@@ -7,12 +7,10 @@ fn next_workspace_number(conn: &mut swayipc::Connection) -> Result<i32, swayipc:
...
+        .map_or(ids.len() as i32 + 1, |w| w[0] + 1))
```
Comment (pblkt) [inline src/main.rs:13]:
```suggestion
        .map_or(ids.len() as i32 + 1, |(cur, _)| cur + 1))
```

File: src/main.rs [hunk 3]
```diff
@@ -25,12 +23,12 @@ fn main() -> Result<(), swayipc::Error> {
         .setting(AppSettings::SubcommandRequired)
         .get_matches();
     let mut conn = swayipc::Connection::new()?;
-    match params.subcommand_name() {
-        Some("open") => {
+    match params.subcommand_name().unwrap() {
```
Comment (pblkt) [inline src/main.rs:26]:
```suggestion
    match params.subcommand_name().expect("unexpected subcommand") {
```

**Comment by nzig** [comment]
@pblkt fixed with itertools

**Comment by pblkt** [comment]
Glorious :crown:

**Pull request merged**